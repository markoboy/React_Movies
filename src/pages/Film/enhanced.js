import {
  moviesStatusSelector,
  selectedMovieSelector,
} from '@store/selectors/moviesSelectors';
import { selectMovie } from '@store/thunks/moviesThunk';
import { connect } from 'react-redux';
import FilmContainer from './container';

const mapStateToProps = (state) => ({
  selectedMovie: selectedMovieSelector(state),
  status: moviesStatusSelector(state),
});

const mapDispatchToProps = {
  selectMovie,
};

const Film = connect(mapStateToProps, mapDispatchToProps)(FilmContainer);

export default Film;
