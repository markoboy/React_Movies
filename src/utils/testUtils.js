/* eslint-disable import/no-extraneous-dependencies */
import defaultStore from '@store/index';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export function renderWithStore(
  ui,
  { store = defaultStore, route = '/', ...renderOptions } = {}
) {
  window.history.pushState({}, 'Test page', route);

  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }

  return render(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

export function renderWithRouter(ui, { route = '/' } = {}) {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: BrowserRouter });
}
