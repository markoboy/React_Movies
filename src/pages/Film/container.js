import { formatMovieData } from '@components/features/ModalForm/utils';
import MovieDetail from '@components/features/MovieDetail';
import { ERROR_STATUS } from '@constants/StatusTypes';
import React, { useEffect, useMemo } from 'react';
import { Redirect, useParams } from 'react-router-dom';

export default function FilmContainer({ selectedMovie, status, selectMovie }) {
  const { id } = useParams();

  useEffect(() => {
    selectMovie(Number(id));
  }, [id]);

  const formattedMovie = useMemo(() => {
    if (selectedMovie) {
      return formatMovieData(selectedMovie);
    }

    return null;
  }, [selectedMovie]);

  return !formattedMovie && status === ERROR_STATUS ? (
    <Redirect to="/not-found" push />
  ) : (
    formattedMovie && <MovieDetail {...formattedMovie} />
  );
}
