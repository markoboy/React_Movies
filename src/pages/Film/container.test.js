import React from 'react';

import { renderWithRouter } from '@utils/testUtils';
import Film from './container';

describe('Film Page', () => {
  it('renders the selected movie when one is selected', () => {
    const selectedMovie = {
      id: 1,
      title: 'A film title',
      poster_path: 'https://film.com/image.jpg',
      overview: 'A great film',
      runtime: 145,
      genres: ['Documentary'],
    };

    const { container } = renderWithRouter(
      <Film selectedMovie={selectedMovie} selectMovie={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });

  it('doesnt render the MovieDetails if no selectedMovie is set', () => {
    const { container } = renderWithRouter(
      <Film selectMovie={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });
});
