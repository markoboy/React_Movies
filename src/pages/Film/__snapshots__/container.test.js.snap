// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film Page doesnt render the MovieDetails if no selectedMovie is set 1`] = `<div />`;

exports[`Film Page renders the selected movie when one is selected 1`] = `
<div>
  <section
    class="section section--large background--dark-black margin--bottom"
  >
    <div
      class="wrapper"
    >
      <div
        class="row"
      >
        <div
          class="column column--m-3"
        >
          <img
            alt="A film title"
            height="500"
            src="https://film.com/image.jpg"
            width="333"
          />
        </div>
        <div
          class="column column--m-9"
        >
          <div
            class="movie-detail"
          >
            <div
              class="movie-detail__heading-container"
            >
              <h1
                class="movie-detail__heading"
              >
                A film title
              </h1>
            </div>
            <p
              class="movie-detail__genre"
            >
              Documentary
            </p>
            <div
              class="movie-detail__year-container"
            >
              <p
                class="movie-detail__duration"
              >
                145 min
              </p>
            </div>
            <p
              class="movie-detail__description"
            >
              A great film
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
